api/src/main/java/org/commonjava/indy/data/DefaultStoreValidator.java

```diff
@@ -0,0 +1,171 @@
+                }
+                // Execute HTTP GET & HEAD requests in separate thread pool from executor service
+                Future<Integer> httpGetStatus = executeGetHttp(new HttpGet(remoteUrl.get().toURI()), httpRequestsLatch);
+                Future<Integer> httpHeadStatus = executeHeadHttp(new HttpHead(remoteUrl.get().toURI()), httpRequestsLatch);
+                // Waiting for Http GET & HEAD Request Executor tasks to finish
+                httpRequestsLatch.await();
+                errors.put(StoreValidationConstants.HTTP_GET_STATUS, httpGetStatus.get().toString());
+                errors.put(StoreValidationConstants.HTTP_HEAD_STATUS, httpHeadStatus.get().toString());
+                if(!remoteUrl.get().getProtocol().equalsIgnoreCase(StoreValidationConstants.HTTPS)) {
+                    errors.put(StoreValidationConstants.HTTP_PROTOCOL, remoteUrl.get().getProtocol());
+                }
+                // Check for Sucessfull Validation
+                if (httpGetStatus.get() < 400 && httpHeadStatus.get() < 400) {
+                    LOGGER.warn("=> Success HTTP GET and HEAD Response from Remote Repository: " + remoteUrl.get());
+                    return new ArtifactStoreValidateData
+                        .Builder(remoteRepository.getKey())
+                        .setRepositoryUrl(remoteUrl.get().toExternalForm())
+                        .setValid(true)
+                        .setErrors(errors)
+                        .build();
+                }
+            }
+        }
+        catch (MalformedURLException mue) {
+            LOGGER.error("=> Mailformed URL: ", mue);
+            errors.put(StoreValidationConstants.MAILFORMED_URL,mue.getMessage());
+            return new ArtifactStoreValidateData
+                .Builder(artifactStore.getKey())
+                .setRepositoryUrl( remoteUrl.get().toExternalForm() )
+                .setErrors(errors)
+                .build();
+        }
+        catch (Exception e) {
+            LOGGER.error(" => Not Valid Remote Repository, \n => Exception: " + e);
+            errors.put(StoreValidationConstants.GENERAL, e.getMessage());
+            return new ArtifactStoreValidateData
+                .Builder(artifactStore.getKey())
+                .setRepositoryUrl( remoteUrl.get().toExternalForm() )
+                .setErrors(errors)
+                .build();
+        }
+        if (artifactStore instanceof RemoteRepository) {
+            return new ArtifactStoreValidateData
+                .Builder(artifactStore.getKey())
+                .setRepositoryUrl(((RemoteRepository) artifactStore).getUrl())
+                .setErrors(errors)
+                .build();
+        }
+        else {
+            return new ArtifactStoreValidateData
+                .Builder(artifactStore.getKey())
+                .build();
+        }
+    }
+
+    private boolean allowedNonSSLHostname(String remoteHost, String host) {
+        return host.contains(remoteHost);
```

Comment:
Could a non-ssl-hosts setting of `.redhat.com` be attacked with `.redhat.composed-of-malware.com` or something?

Reply:
I was thinking for this one that best would be if we enter regexes in allowed hostnames, because if we wanna to be general ( like everything from domain  .redhat.com is allowed ) that would be best case scenario , other thing ( like brute force separating on "." and then comparing that value and then searching for other value and then if it is star then allow anything that follow ... ) is more prone to errors.

Reply:
We don't have anything like that yet, we'd have to build it if we wanted it. Regex is more dependable from a code point of view, but more error-prone from a configuration point of view (they're much easier to get wrong). I guess let's stick with regex for now...we have this in other places, and at least it's well-understood.